src/Bellhop.js

```diff
@@ -269,11 +269,17 @@ export class Bellhop extends BellhopEventDispatcher {
    *  @param {Boolean} [runOnce=false] If we only want to respond once and then remove the listener
    */
   respond(event, data = {}, runOnce = false) {
-    const internalCallback = e => {
+    const internalCallback = async function(e){
       if (runOnce) {
         this.off(e.type, internalCallback);
       }
-      this.send(event, 'function' === typeof data ? data() : data);
+      if(typeof data === 'function'){
+        data = data(); 
+      }
+      if(typeof data === 'Promise'){
```

Comment:
I am certain that this doesn't work, as Promise isn't a primitive type. The good news is that `data = await 0;` and `data = await Promise.resolve(0);` both work - you can await non-promises just fine.

Reply:
Also `instanceof` would probably work fine here. But I agree, `await nonPromise` reads fine and works fine too.